build.gradle

```diff
@@ -355,44 +355,38 @@ project(':iceberg-pig') {
 project(':iceberg-runtime') {
   apply plugin: 'com.github.johnrengelman.shadow'
 
-  configurations {
-    shadow
-    compileOnly.extendsFrom shadow
-  }
-
-  shadowJar {
-    configurations = [project.configurations.shadow]
-    from(project.sourceSets.main.output)
-  }
-
-  tasks.build.dependsOn tasks.shadowJar
+  tasks.assemble.dependsOn tasks.shadowJar
   tasks.install.dependsOn tasks.shadowJar
   tasks.javadocJar.dependsOn tasks.shadowJar
 
   dependencies {
-    shadow project(':iceberg-api')
-    shadow project(':iceberg-common')
-    shadow project(':iceberg-core')
-    shadow project(':iceberg-parquet')
-    shadow project(':iceberg-spark')
-    shadow project(':iceberg-pig')
-    shadow project(':iceberg-hive')
-
-    shadow "org.apache.avro:avro:$avroVersion"
-    shadow "org.apache.parquet:parquet-avro:$parquetVersion"
+    compile project(':iceberg-api')
+    compile project(':iceberg-common')
+    compile project(':iceberg-core')
+    compile project(':iceberg-parquet')
+    compile project(':iceberg-spark')
+    compile project(':iceberg-pig')
+    compile project(':iceberg-hive')
+
+    compile "org.apache.avro:avro:$avroVersion"
+    compile "org.apache.parquet:parquet-avro:$parquetVersion"
+
   }
 
   publishing {
     publications {
       withType(IvyPublication) {
-        configurations {
-          'shadow' { extend 'runtime' }
-        }
+        from components.java
       }
     }
   }
 
   shadowJar {
+    dependencies {
```

Comment:
As discussed in the mail thread, we should probably relocate guava also here and change it to be at compile scope?